layers/cars.py

```diff
@@ -0,0 +1,153 @@
+
+#Below code not working - cannot find file (althugh URL work in a web browser)
+#To work around download file from below url and save in WORKING_DIR
+
+# CENSUS_URL = "https://static.ons.gov.uk/datasets/a20437fb-ae7f-439b-bc91-de261335038b/TS045-2021-3-filtered-2023-03-13T16:49:47Z.csv"
+# download_vehicle_ownership(CENSUS_URL, WORKING_DIR)
+
+carOwnerFile = 'TS045-2021-3-filtered-2023-03-13T16 49 47Z.csv'
+
+#%% Read data in
+
+#OAs shapefile as geopandas dataframe
+oas = gpd.read_file(output_areas_geojson_path)
+#Car ownership data
+carsRaw = pd.read_csv('{}/{}'.format(WORKING_DIR,carOwnerFile))
+
+#%% Cars
+
+#Empty dataframe to capture data
+cars = pd.DataFrame(columns = ['0 Cars', '1 Car', '2 Car', '3+ Cars', 'Total'], index = carsRaw['Output Areas Code'].value_counts().index)
+
+#Split data from cars ownership across different columns
+cars['0 Cars'] = carsRaw[carsRaw['Car or van availability (5 categories) Code'] == 0][['Output Areas Code','Observation']].set_index('Output Areas Code')
+cars['1 Car'] = carsRaw[carsRaw['Car or van availability (5 categories) Code'] == 1][['Output Areas Code','Observation']].set_index('Output Areas Code')
+cars['2 Car'] = carsRaw[carsRaw['Car or van availability (5 categories) Code'] == 2][['Output Areas Code','Observation']].set_index('Output Areas Code')
+cars['3+ Cars'] = carsRaw[carsRaw['Car or van availability (5 categories) Code'] == 3][['Output Areas Code','Observation']].set_index('Output Areas Code')
+#Total number of cars per OA
+cars['Total'] = carsRaw.groupby('Output Areas Code').sum()['Observation']
+#Percentage of households with a car
+cars['Pcnt HH With Car'] = 1 - (cars['0 Cars'] / cars['Total'])
+#Weighted average cars per hh
+cars['Avg Cars per HH'] = (cars['1 Car'] + (cars['2 Car'] * 2) + (cars['3+ Cars'] * 3)) / cars['Total']
+
+#%% Banding Percent HH with Car
+
+conditionsPcHH = [
+(cars['Pcnt HH With Car'] <= 0.4),
+(cars['Pcnt HH With Car'] <= 0.6),
+(cars['Pcnt HH With Car'] <= 0.7),
+(cars['Pcnt HH With Car'] <= 0.8),
+(cars['Pcnt HH With Car'] <= 0.85),
+(cars['Pcnt HH With Car'] <= 0.9),
+(cars['Pcnt HH With Car'] <= 0.95),
+(cars['Pcnt HH With Car'] <= 1)]
+
+choicesPcHH = [1,2,3,4,5,6,7,8]
+cars['Pcnt HH With Car Bands'] = np.select(conditionsPcHH, choicesPcHH, default=99)
+print(cars['Pcnt HH With Car Bands'].value_counts())
+
+#%% Banding weighted average cars per HH
+conditionsNumCars = [
+(cars['Avg Cars per HH'] <= 0.5),
+(cars['Avg Cars per HH'] <= 0.75),
+(cars['Avg Cars per HH'] <= 1),
+(cars['Avg Cars per HH'] <= 1.25),
+(cars['Avg Cars per HH'] <= 1.5),
+(cars['Avg Cars per HH'] <= 1.75)]
+
+choicesNumCars = [1,2,3,4,5,6]
+cars['Avg Cars per HH Bands'] = np.select(conditionsNumCars, choicesNumCars, default=7)
+print(cars['Avg Cars per HH Bands'].value_counts())
+
+#%% Output data
+output_file = 'car-ownership-layer.csv'
+cars = cars.merge(oas[['OA21CD','geometry']],left_index = True, right_on = 'OA21CD')
+cars.to_csv('{}/{}'.format(WORKING_DIR,output_file))
```

Comment:
Can we output as a GeoJSON file instead? That'll make it easier to load in the map, and I think geopandas can do it easily

Reply:
Changed script to output as geojson